sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets

```diff
@@ -11,211 +11,239 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
+    <UsingTask TaskName="GenerateFunctionMetadata"
+             AssemblyFile="$(_FunctionsTaskAssemblyFullPath)"/>
+
+    <UsingTask TaskName="CreateZipFileTask"
+             AssemblyFile="$(_FunctionsTaskAssemblyFullPath)"/>
+
+    <UsingTask TaskName="ZipDeployTask"
+              AssemblyFile="$(_FunctionsTaskAssemblyFullPath)"/>
+
+    <Import Project="$(MSBuildFunctionsTargetsPath)Microsoft.Azure.Functions.Designtime.targets"
+           Condition="Exists('$(MSBuildFunctionsTargetsPath)Microsoft.Azure.Functions.Designtime.targets')" />
+
+    <Target Name="_FunctionsPreBuild" BeforeTargets="BeforeBuild">
+        <Message Condition="'$(_AzureFunctionsNotSet)' == 'true'" Importance="high" Text="AzureFunctionsVersion not configured in the project. Setting AzureFunctionsVersion to v3"/>
+        <Error Condition="$(AzureFunctionsVersion.StartsWith('v1',StringComparison.OrdinalIgnoreCase))" Text="AzureFunctionsVersion is set to an incompatible version, Please set it to v4"/>
+        <Error Condition="$(AzureFunctionsVersion.StartsWith('v2',StringComparison.OrdinalIgnoreCase))" Text="AzureFunctionsVersion is set to an incompatible version, Please set it to v3"/>
+        <Error Condition="!$(AzureFunctionsVersion.StartsWith('v3',StringComparison.OrdinalIgnoreCase)) And !$(AzureFunctionsVersion.StartsWith('v4',StringComparison.OrdinalIgnoreCase))" Text="AzureFunctionsVersion is set to an incompatible version"/>
+        <Error Condition="'$(_ToolingSuffix)' == ''" Text="Invalid combination of TargetFramework and AzureFunctionsVersion is set."/>
+        <Error Condition="'$(_IsFunctionsSdkBuild)' == 'true'" Text="Microsoft.NET.Sdk.Functions package is meant to be used with in-proc function apps. Please remove the reference to this package in isolated function apps."/>
+    </Target>
+
+    <Target Name="_FunctionsPostBuild" AfterTargets="AfterBuild" DependsOnTargets="_FunctionsPostBuildNetFx;_FunctionsPostBuildNetApp"/>
+    <!--.NET Framework post build-->
+    <Target Name="_FunctionsPostBuildNetFx" Condition="$(TargetFrameworkIdentifier) == '.NETFramework'">
+        <PropertyGroup>
+            <OutputFile>$(TargetDir)\worker.config.json</OutputFile>
+            <ExtensionsCsProjFilePath>$([System.IO.Path]::Combine($([System.IO.Path]::GetTempPath()), $([System.IO.Path]::GetRandomFileName())))</ExtensionsCsProjFilePath>
+        </PropertyGroup>
+
+        <GenerateFunctionMetadata
+          AssemblyPath="$(TargetDir)$(AssemblyName).exe"
+          ReferencePaths="@(ReferencePath)"
+          ExtensionsCsProjFilePath="$(ExtensionsCsProjFilePath)"
+          AzureFunctionsVersion="$(AzureFunctionsVersion)"
+          OutputPath="$(TargetDir)"/>
+
+        <Copy
+          SourceFiles="$(_FunctionsMetadataLoaderExtensionFile)"
+          DestinationFolder="$(ExtensionsCsProjFilePath)\buildout"
+          OverwriteReadOnlyFiles="true" />
+
+        <WriteLinesToFile
+            Condition="$(TargetFrameworkIdentifier) == '.NETFramework'"
```

Comment:
Should we remove the `Condition` expression as it is already in the parent level (line 60)?

Reply:
👍🏼